examples/credential-registry/Cargo.toml

```diff
@@ -0,0 +1,30 @@
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[package]
+name = "credential_registry"
+version = "0.1.0"
+edition = "2018"
+license = "MPL-2.0"
+authors = [ "Danil Annenkov <danil.v.annenkov@gmail.com>" ]
+description = "An example credential registry smart contract"
+
+[features]
+default = ["std", "crypto-primitives", "wee_alloc"]
+std = ["concordium-std/std", "concordium-cis2/std"]
+wee_alloc = ["concordium-std/wee_alloc"]
+crypto-primitives = ["concordium-std/crypto-primitives"]
+
+[dependencies]
+concordium-std = {path = "../../concordium-std", default-features = false}
```

Comment:
I thought that this line

```toml
[features]
# ...
crypto-primitives = ["concordium-std/crypto-primitives"]
```

enables `crypto-primitives`. Maybe I don't understand how it works, could you please explain?

Reply:
This line is adding a feature to your smart contract project, which (when enabled) will enable the `"crypto-primitives"` feature in `concordium-std`. 
Your smart contract feature is enabled by default, so running `cargo build` works, but if you run `cargo build --no-default-features` it will be disabled and fail to compile.

Reply:
The features defined in `[features]` are build options for the user, so they can enable/disable features to reduce compile time, but since this project will only work if it is enabled, this should not be a choice of the user.

Reply:
I've added versions as @abizjak suggested and kept the rest as it is.